src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
+
+        $now = DBDatetime::now();
+        $lastEditedSentinel = $now
+            // This is important to set otherwise it will break unit tests due to how the date mock works
+            ->setImmutable(true)
+            ->modify(sprintf('-%d minutes', $jobRetrySentinel))
+            ->Rfc2822();
+
+        $autoRetryJobs = QueuedJobDescriptor::get()
+            ->filter([
+                // Automatic retry is applied to only configured jobs statuses
+                'JobStatus' => $jobStatusConditions,
+                // Make sure to avoid retrying jobs that are being processed very recently to avoid edge cases
+                'LastEdited:LessThan' => $lastEditedSentinel,
+                // We only want to process jobs in the current queue
+                'JobType' => $queueType,
+            ])
+            // Oldest jobs first
+            ->sort('ID', 'ASC');
+
+        // Provide an option to customise the list of jobs that are eligible for a job retry
+        $this->extend('updateRetryJobsList', $autoRetryJobs);
+
+        $jobConfigs = [];
+        $jobClasses = $autoRetryJobs->columnUnique('Implementation');
+
+        foreach ($jobClasses as $jobClass) {
+            // Skip invalid job data (orphaned data)
+            if (!class_exists($jobClass)) {
+                continue;
+            }
+
+            $maxRetryAttempts = (int) Config::inst()->get($jobClass, 'max_retry_attempts');
+
+            // Job is not configured to be retried
+            if ($maxRetryAttempts <= 0) {
+                continue;
+            }
+
+            $initialRetryDelay = (int) Config::inst()->get($jobClass, 'initial_retry_delay');
+            $retryFalloffMultiplier = (float) Config::inst()->get($jobClass, 'retry_falloff_multiplier');
+            $retryFalloffMultiplierVariance = (float) Config::inst()->get(
+                $jobClass,
+                'retry_falloff_multiplier_variance'
+            );
+
+            // Sanitise values so we don't have to deal with edge cases
+            $initialRetryDelay = max(0, $initialRetryDelay);
+            $retryFalloffMultiplier = max(1, $retryFalloffMultiplier);
+            $retryFalloffMultiplierVariance = max(0, $retryFalloffMultiplierVariance);
+
+            $jobConfigs[$jobClass] = [
+                $maxRetryAttempts,
+                $initialRetryDelay,
+                $retryFalloffMultiplier,
+                $retryFalloffMultiplierVariance,
+            ];
+        }
+
+        // We couldn't find any job candidates to retry
+        if (!$jobConfigs) {
+            return;
+        }
+
+        $retryJobIDs = [];
+        $remainingLimit = $jobRetryLimit;
+
+        foreach ($jobConfigs as $jobClass => $retryConfig) {
+            // We've reached the number of jobs we are allowed to retry in this health check run
+            if ($remainingLimit <= 0) {
+                break;
+            }
+
+            [
+                $maxRetryAttempts,
+            ] = $retryConfig;
```

Comment:
```suggestion
            [$maxRetryAttempts] = $retryConfig;
```
Multi-line arrays are only used when there are multiple values (or in extreme cases if the single value is really really long). We should apply the same logic to this array destructuring syntax.

That said, it's probably cleaner to use associative arrays instead of destructuring an indexed array. It also removes the need to use destructuring again further down.

Reply:
I use multi-line syntax as this produces cleaner diffs in the future. Will keep this as is.